promises/test.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+import { assert, test } from "../testing/mod.ts";
+import { Deferred, defer, delay, timeout, TimeoutError } from "./mod.ts";
+
+function isDeferred(x): x is Deferred {
+  return (
+    typeof x === "object" &&
+    x.promise instanceof Promise &&
+    typeof x["resolve"] === "function" &&
+    typeof x["reject"] === "function"
+  );
+}
+
+async function timeTaken<T>(f: () => Promise<T>): Promise<number> {
+  const before = new Date().getTime(); // Performance.now();
+  await f();
+  const after = new Date().getTime(); // Performance.now();
+  return after - before;
+}
+
+test(async function asyncIsDeferred() {
+  const d = defer();
+  assert.assert(isDeferred(d));
+  assert.assert(
+    isDeferred({
+      promise: null,
+      resolve: () => {},
+      reject: () => {}
+    }) === false
+  );
+});
+
+test(async function delayed() {
+  const t = await timeTaken(async () => await delay(200));
+  assert(t > 100);
```

Comment:
just a nit-pick, isn't it guaranteed that timeout will take at least 200 ms?

Reply:
Should be! I was just being cautious here... I think it's _probably_ okay to decrease 😬 

All these tests pass if I use >= 200 (on each of them).